utils/ccloud_library.sh

```diff
@@ -731,17 +731,27 @@ function ccloud::set_kafka_cluster_use() {
 function ccloud::create_ccloud_stack() {
   enable_ksql=$1
 
-  RANDOM_NUM=$((1 + RANDOM % 1000000))
-  #echo "RANDOM_NUM: $RANDOM_NUM"
+  if [[ -z "$SERVICE_ACCOUNT_ID" ]]; then
+    # Service Account is not received so it will be created
+    local RANDOM_NUM=$((1 + RANDOM % 1000000))
+    SERVICE_NAME=${SERVICE_NAME:-"demo-app-$RANDOM_NUM"}
+    SERVICE_ACCOUNT_ID=$(ccloud::create_service_account $SERVICE_NAME)
+  fi
 
-  SERVICE_NAME="demo-app-$RANDOM_NUM"
-  SERVICE_ACCOUNT_ID=$(ccloud::create_service_account $SERVICE_NAME)
-  echo "Creating Confluent Cloud stack for new service account id $SERVICE_ACCOUNT_ID of name $SERVICE_NAME"
+  if [[ "$SERVICE_NAME" == "" ]]; then
+    echo "ERROR: SERVICE_NAME is not defined. If you are providing the SERVICE_ACCOUNT_ID to this function please also provide the SERVICE_NAME"
+    exit 1
+  fi
 
-  ENVIRONMENT_NAME="demo-env-$SERVICE_ACCOUNT_ID"
-  ENVIRONMENT=$(ccloud::create_and_use_environment $ENVIRONMENT_NAME)
+  echo "Creating Confluent Cloud stack for service account $SERVICE_NAME, ID: $SERVICE_ACCOUNT_ID."
 
-  CLUSTER_NAME=demo-kafka-cluster-$SERVICE_ACCOUNT_ID
+  if [[ -z "$ENVIRONMENT" ]]; then
+    # Environment is not received so it will be created
+    ENVIRONMENT_NAME=${ENVIRONMENT_NAME:-"demo-env-$SERVICE_ACCOUNT_ID"}
+    ENVIRONMENT=$(ccloud::create_and_use_environment $ENVIRONMENT_NAME) 
+  fi
+  
+  CLUSTER_NAME=${CLUSTER_NAME:-"demo-kafka-cluster-$SERVICE_ACCOUNT_ID"}
   CLUSTER_CLOUD="${CLUSTER_CLOUD:-aws}"
   CLUSTER_REGION="${CLUSTER_REGION:-us-west-2}"
   CLUSTER=$(ccloud::create_and_use_cluster $CLUSTER_NAME $CLUSTER_CLOUD $CLUSTER_REGION)
```

Comment:
@gianlucanatali can we add this change please, it's rquired for cluster names with spaces.  Thank you

```suggestion
  CLUSTER=$(ccloud::create_and_use_cluster "$CLUSTER_NAME" $CLUSTER_CLOUD $CLUSTER_REGION)
```